### PR TITLE
Extract meta inf

### PR DESCRIPTION
**Pull request by cccs-dawson**
Changes to Espresso module:
	--a new results section is added to the report for manifest and certificate data
	--the main class and main package are included as tags in a subsection--see [AL common changes](https://github.com/CybercentreCanada/assemblyline-base/pull/169)
	--Use keytool to printcert. There is a subsection for each certificate. some meta info from the certificates(s) are included as tags. The entire certificate is dumped into the body.
	--some simple heuristics are also included based on the certificate metadata--the heuristics are similar to those used in APKaye
	--add some simple tests to test keytool_parse module (which is installed on a per service basis) and to test Espresso.execute. See test folder.

### CODE REVIEW

File: espresso.py [hunk 1]
```diff
@@ -333,14 +453,13 @@ def execute(self, request):
             for embed in new_files:
                 request.add_extracted(embed, embed.replace(extract_dir + "/", "").replace(decompiled_dir + "/", ""),
                                       txt)
+        # for path, name, desc in supplementary_files:
+        #     print(f'---{name}: {type(path)}, {type(name)}, {type(desc)}')
```
Comment (cccs-sgaron) [inline espresso.py:457]: Remove debug code

File: espresso.py [hunk 2]
```diff
@@ -210,6 +218,100 @@ def analyse_class_file(self, file_res, cf, cur_file, cur_file_path, start_bytes,
...
+                            elif len(main) == 1:
+                                self.manifest_tags.append(('file.jar.main_class', main[0]))
+
+                        # elif f.startswith(b'MIDlet'):
+                        #     if f.endswith(b'Name'):
+                        #         self.manifest_tags.append( ('file.jar.midlet.name', v) )
+                        #     elif re.search(b'\d+$', f):
+                        #         midlet_n = v.split(b',')
+                        #         self.manifest_tags.append( ('file.jar.midlet.midlet_name', midlet_n[0]) )
+                        #         self.manifest_tags.append( ('file.jar.midlet.main_class', midlet_n[2]) )
```
Comment (cccs-sgaron) [inline espresso.py:307]: Remove unused code

File: espresso.py [hunk 3]
```diff
@@ -2,11 +2,14 @@
 import logging
 import os
 import zipfile
-from subprocess import PIPE, Popen
+import time
+from subprocess import Popen, PIPE, call
+import re
 
+from assemblyline.common.keytool_parse import Certificate, certificate_chain_from_printcert, keytool_printcert
```
Comment (cccs-sgaron) [inline espresso.py:9]: Keytool parse should be move to assemblyline_v4_service

**Review by cccs-sgaron — CHANGES_REQUESTED** [review]

**Comment by cccs-sgaron** [comment]
We probably have to install keytool during the docker container creation, we are using a very minimal base image...

**Comment by cccs-rs** [comment]
We'll want to move the keytool_parse module to be used locally instead of part of the service base:
https://github.com/CybercentreCanada/assemblyline-v4-service/pull/35

**Pull request merged**